src/v/v8_engine/script_dispatcher.h

```diff
@@ -0,0 +1,146 @@
+          .second;
+    }
+
+    bool remove(const model::topic_namespace& topic) {
+        bool contains_script_flag = _compilation_params.erase(topic) == 1;
+        if (!contains_script_flag) {
+            return false;
+        }
+
+        auto runtime = _runtimes.find(topic);
+        if (runtime == _runtimes.end()) {
+            return true;
+        }
+
+        if (runtime->second.get()->function.get() == nullptr) {
+            runtime->second->is_deleted = true;
+            runtime->second->cv.broadcast();
+        }
+
+        _runtimes.erase(runtime);
+
+        return true;
+    }
+
+    ss::future<ss::lw_shared_ptr<script>> get(model::topic_namespace topic) {
+        auto params_it = _compilation_params.find(topic);
+        if (params_it == _compilation_params.end()) {
+            co_return ss::lw_shared_ptr<script>(nullptr);
+        }
+
+        auto runtime_it = _runtimes.find(topic);
+        if (runtime_it == _runtimes.end()) {
+            auto value = ss::make_lw_shared(runtimes_value());
+            _runtimes.emplace(value);
+
+            auto runtime = ss::make_lw_shared(script(
+              params_it->second.max_heap_size_bytes,
+              params_it->second.run_timeout_ms));
+
+            try {
+                co_await runtime->init(
+                  params_it->second.function_name,
+                  std::move(params_it->second.code),
+                  _executor);
+            } catch (const v8_engine::script_exception& ex) {
+                // add log
+                if (!value->is_deleted) {
+                    // Need to delete runtime from hash_map
+                    _runtimes.erase(topic);
+                    value->cv.broadcast();
+                    co_return nullptr;
+                }
+            }
+
+            // User delete runtime, and we can skip it
+            if (value->is_deleted) {
+                co_return nullptr;
+            }
+
+            value->function = runtime;
+            value->cv.broadcast();
+            co_return runtime;
+        } else {
+            auto value = runtime_it->second;
+            if (value->function.get() != nullptr) {
+                co_return value->function;
+            }
+
+            co_await value->cv.wait();
+            co_return value->function;
+        }
+    }
+
+private:
+    absl::node_hash_map<model::topic_namespace, script_compile_params>
+      _compilation_params;
+
+    struct runtimes_value {
+        ss::condition_variable cv;
+        ss::lw_shared_ptr<script> function{nullptr};
+        // We need to understand do user delete current dp or not
+        bool is_deleted{false};
+    };
+
+    // Hash map from topic to runtime. We need to store shared_ptr for value,
+    // because delete can remove it from hash_map while get will work with this
+    // value
+    absl::
+      node_hash_map<model::topic_namespace, ss::lw_shared_ptr<runtimes_value>>
```

Comment:
what is the difference between a `runtime`, `runtime_value`, and a `script`? is it possible to unify the terminology?